lib/puppet/provider/kubeconfig/ruby.rb

```diff
@@ -40,6 +42,13 @@ def chown
     FileUtils.chown(resource[:owner], resource[:group], resource[:path])
   end
 
+  def chmod
+    mode = resource[:mode]
+    mode = mode.to_i(8) if mode.is_a?(String) && mode =~ %r{^[0-7]+$}
```

Comment:
should we ensure we have 4 digits?

Reply:
The munging and validation should only result in valid numerical strings, but I guess it doesn't hurt to make the validation stricter here as well.